x/profile/keeper/msg_server_delete_coordinator.go

```diff
@@ -0,0 +1,28 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/tendermint/spn/x/profile/types"
+)
+
+func (k msgServer) DeleteCoordinator(
+	goCtx context.Context,
+	msg *types.MsgDeleteCoordinator,
+) (*types.MsgDeleteCoordinatorResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Check if the coordinator address is already in the store
+	coordByAddress, found := k.GetCoordinatorByAddress(ctx, msg.Address)
+	if !found {
+		return &types.MsgDeleteCoordinatorResponse{},
+			sdkerrors.Wrap(types.ErrCoordAddressNotFound, msg.Address)
+	}
+
+	coord := k.GetCoordinator(ctx, coordByAddress.CoordinatorId)
```

Comment:
We already have the coordinator ID from `GetCoordinatorByAddress`

Reply:
We can actually let this line though, it wasn't mentioned in the specs but wecan do the same as `UpdateCoordinatorDescription`: if the coordinator doesn't exist while the address is registered, we panic